### ISSUE TITLE
Unary minus operator at beginning of expression is ignored

### ISSUE THREAD

**Issue by zwebert31** [HubSpot/jinjava#892]
Adding parenthesis around an expression with a unary sub-expression, changes the output incorrectly.

`{{ (-10 | abs + 4)}}  is -6`
`{{ -10 | abs + 4 }}  is 14`

In the former example, the unary operation is being applied last, whereas in the latter example, the unary operation is being applied before the `abs` filter (as i would expect)

**Comment by tkindy** [comment]
Just to check my understanding, I confirmed that Jinja handles this as I would expect.

![jinja examples](https://user-images.githubusercontent.com/2991842/179812690-6b7755c8-2004-4e24-bfd9-b37158ae07d5.png)

**Comment by tkindy** [comment]
I believe I've found the bug, and it's even more basic than the examples so far in this issue. The template `{{ -10 }}` parses as `-10` in Jinja and `10` in Jinjava.

The bug is coming from a feature of Jinjava (which, as expected, comes from Jinja) called [whitespace control](https://jinja.palletsprojects.com/en/3.1.x/templates/#whitespace-control). In an expression or other block, you can add a dash at the beginning or end to control whether preceding or trailing whitespace on the line is trimmed. For example, the template `{{- "foo" }}` would trim all leading whitespace on the line ahead of `foo`.

The Jinja docs specify, "You must not add whitespace between the tag and the minus sign." However, the current Jinjava implementation [allows whitespace in between the tag and the dash](https://github.com/HubSpot/jinjava/blob/8882c8f4474995bc8730acdbfe3cebcab185f46f/src/main/java/com/hubspot/jinjava/tree/parse/ExpressionToken.java#L48-L51), so negative numbers at the start of expressions are having their minus sign interpreted as a whitespace control character by accident.

This is also the case in statements (`{%`), though I don't know of a valid statement off the top of my head that could start with a negative number.

I should be able to fix this relatively easily.

**Comment by tkindy** [comment]
I've managed to fix the whitespace control problem locally, but it's revealed another issue with operator precedence. Specifically, the unary minus should bind tighter than operators like `is`; that is, `-1 is integer` should evaluate the same as `(-1) is integer`. However, currently Jinjava evaluates this as `-(1 is integer)` which is obviously incorrect.

This is a tougher issue to untangle, but I did a fair amount of digging yesterday and hope to make some more progress today.
